domain/main.tf

```diff
@@ -1,53 +1,55 @@
-  service_name = (var.name == "" ? "${data.cloudfoundry_app.app[local.target_apps[0]].name}-${var.domain_name}" : var.name)
-}
-
-data "cloudfoundry_space" "space" {
-  org_name = var.cf_org_name
-  name     = var.cf_space_name
+  service_name  = (var.name == "" ? "${var.app_names[0]}-${var.domain_name}" : var.name)
+  tags          = setunion(["terraform-cloudgov"], var.tags)
+  connect_route = length(var.app_names) > 0
+  endpoint      = (local.connect_route ? cloudfoundry_route.origin_route_connected.0.url : cloudfoundry_route.origin_route.0.url)
 }
 
 data "cloudfoundry_app" "app" {
-  for_each   = toset(local.target_apps)
-  name_or_id = each.key
-  space      = data.cloudfoundry_space.space.id
+  for_each   = toset(var.app_names)
+  name       = each.key
+  space_name = var.cf_space.name
+  org_name   = var.cf_org_name
 }
 
 ###########################################################################
 # There are two prerequisites for running this module:
 #
 # 1) Domain must be manually created by an OrgManager:
-#     cf create-domain <%= cloud_gov_organization %> TKTK-production-domain-name
+#     cf create-domain <%= var.cf_org_name %> <%= var.domain_name %>
 # 2) ACME challenge record must be created.
 #     See https://cloud.gov/docs/services/external-domain-service/#how-to-create-an-instance-of-this-service
 ###########################################################################
 data "cloudfoundry_domain" "origin_url" {
   name = var.domain_name
 }
 
-resource "cloudfoundry_route" "origin_route" {
-  domain   = data.cloudfoundry_domain.origin_url.id
-  hostname = var.host_name
-  space    = data.cloudfoundry_space.space.id
+resource "cloudfoundry_route" "origin_route_connected" {
+  count  = local.connect_route ? 1 : 0
+  space  = var.cf_space.id
+  domain = data.cloudfoundry_domain.origin_url.id
+  host   = var.host_name
 
-  dynamic "target" {
-    for_each = data.cloudfoundry_app.app
-    content {
-      app = target.value.id
-    }
-  }
+  destinations = [for name, app in data.cloudfoundry_app.app : { app_id = app.id }]
+}
+
+resource "cloudfoundry_route" "origin_route" {
+  count  = local.connect_route ? 0 : 1
+  space  = var.cf_space.id
+  domain = data.cloudfoundry_domain.origin_url.id
+  host   = var.host_name
 }
 
-data "cloudfoundry_service" "external_domain" {
-  name = "external-domain"
+data "cloudfoundry_service_plans" "external_domain" {
+  name                  = var.cdn_plan_name
+  service_offering_name = "external-domain"
 }
 
 resource "cloudfoundry_service_instance" "external_domain_instance" {
   name         = local.service_name
-  space        = data.cloudfoundry_space.space.id
-  service_plan = data.cloudfoundry_service.external_domain.service_plans[var.cdn_plan_name]
-  json_params  = "{\"domains\": \"${local.endpoint}\"}"
-  tags         = var.tags
+  space        = var.cf_space.id
+  service_plan = data.cloudfoundry_service_plans.external_domain.service_plans.0.id
```

Comment:
Shouldn't `var.cdn_plan_name` be used to index into `service_plans` here?

Reply:
Never mind, I just learned that supplying `name = var.cdn_plan_name` to `cloudfoundry_service_plans` will filter the result to just that one service plan, so there will only be one result and `.0.` is always right!